ABC -- ( Allele-specific Binding from ChIP-Seq ) 

Identifies potential allele-specific binding events at known heterozygous positions within the aligned reads of a ChiP-Seq experiment. 
ABC requires at a minimum two (2) files a sorted .sam file of a ChIP-Seq experiment and a file containing the position, strand and allele
information of heterozygous Single Nucleotide Variants (SNVs), either SNPs and/or Mutations. ABC calls allele-specific binding by identifying a bias in
the distribution of the SNV alleles while attempting to control for potential false positives. If you have genomic sequence data
you can use the allele ratio in the DNA as the expected frequency to control for chromosome copy number.

CAUTION:

Care should be taken in the selection of SNVs. Homozygous SNVs or SNVs within duplications will appear to have strong allele-specific effects. 
It is recommended that you filter SNPs prior to running ABC (ie. those mapping to a motif).

Usage: perl ABC.pl --sam <input.sam> --snv-file <snv filename> --out <output filename>

--sam (Required)
		Specify ChIP-Seq .sam file.
		Note: the .sam file must be sorted

--bg (Optional)
		Specify a .bedgraph file capturing the ChIP-Seq signal (shifted read pileups) of the .sam file
		This can be useful to prioritize SNPs with low coverage, since they may fall within centre of the +ve and -ve strand peaks.
		This is caused by short reads and is not necessary longer reads.

--snv-file (Required)
		A tab-delimited text file containing the list of heterozygous SNPs.

                The format of the SNV file is as follows (Do not include the header):

	Example:	SNV_ID	CHR	POSITION	STRAND	REF_ALLELE	OBSERVED_ALLELES	ALLELE_RATIO_DNA
           
	        	rs111	chr1	1111	+	A	A/C	0.5
                	SNV1	chr10	10234	-	C	C/G	0.4

		It is the responsibility of the user to verify the quality of the SNVs (ie Do they pass Hardy-Weinberg equilibrium, etc...).

--out (Optional)
		Specify the ouput file prefix (default ABC).

		(ie. ABC will create two output files ABC.dist and ABC.align)

--min-reads (Optional)
		The minimum number of reads covering the a SNVs (default: 25).
		(ie. ABC will report only those SNPs/Mutations with # reads overlapping them.)

--d (Optional)
		Divide chromosomes in d segments for faster retrieval (default: 2000). 
                A large number of SNVs can take a long time to process. You may try to increase the value of d.
		However, very large numbers will not necessarily increase speed.

--mw-thres (Optional)
		P-value threshold for the Mann-Whitney test used to test a bias in the read position between the SNV alleles. (default: 0.05)
		To report all SNVs set this parameter to 0.

--f-thres (Optional) 
		P-value threshold for the Fisher's exact test used to test for a bias between the strand distribution of the SNV alleles. (default: 0.05)
                To report all SNVs set this parameter to 0.

--verbose (Optional)
		Print progress and SNV results summary to screen.	

--help		
		Prints command line options


		***	VISUALIZING SNV RESULTS	***

		Create a figure of the distribution of reads containing a SNV of interest

		This step requires that the ABC has finished running.
		Once ABC is finished a figure can be generated by specifying the output file prefix used in the initial run and the SNV ID as follows:

		perl ABC.pl --out <output filename prefix> --visualize-snv <SNV ID>




***** OUTPUT ****

.dist		Results Table
		
		SNV		SNV Identifier
		CHR		Chromosome
		BP		Position on chromosome
		REF		Reference allele
		OBS		Observed alleles
		A1		Allele 1
		N_A1		Number of reads containing A1
		F_A1		Frequency of A1
		N_A1_POS	Number of reads aligned to the +ve strand containing A1
		N_A1_NEG	Number of reads aligned to the -ve strand containing A1
		A2		Allele 2 
		N_A2		Number of reads containing A2
		F_A2		Frequency of A2
		N_A2_POS	Number of reads aligned to the +ve strand containing A2
		N_A2_NEG	Number of reads aligned to the -ve strand containing A2
		N_TOTAL		Sum of N_A1 and N_A2
		N_ERRORS	Number of reads containing a nucleotide other than A1 or A2
		MAX		The maximum value of from the .bedgraph file (if specified) within the 2x read length window of the SNV
		P_BINOM		The P-value used to call allele-specific binding
		P_MANN_WHIT	The P-value used to call a bias in read position between alleles 
		P_FISHER	The P-value used to call a bias in the strand distribution between alleles
		P_CHISQ	 	(Same as P_FISHER greater number of reads required)
		P_STRAND	A binomial p-value of the differences in strand abundance 
		A1_Position	Position of A1 within reads
		A1_Strand	Strand of reads containing A1
		A2_Position	Position of A2 within reads 
		A2_strand	Strand of reads containing A2

.align		Contains the alignments separated by Allele for each SNV
